sleap/gui/widgets/video.py

```diff
@@ -1732,6 +1733,211 @@ def updateEdge(self, node: QtNode):
+        # Scale the bounding box and QtInstance if an edge box is selected
+        if event.button() == Qt.RightButton:
+            pass
+        else:
+            scaling_buffer = 10
+
+            x1, y1, x2, y2 = self.rect().getCoords()
+            new_x = event.pos().x()
+            new_y = event.pos().y()
+
+            w = self.parent.player.video.width
+            h = self.parent.player.video.height
+
+            if self.resizing == "top_left":
+                # Check to see if outside the range of the original bounding box
+                if new_x < 0:
+                    new_x = 0
+                if new_x >= x2 - scaling_buffer - self.box_width:
+                    new_x = x2 - scaling_buffer - self.box_width
+                if new_y < 0:
+                    new_y = 0
+                if new_y >= y2 - scaling_buffer - self.box_width:
+                    new_y = y2 - scaling_buffer - self.box_width
+
+                # Update the bounding box
+                self.setRect(QRectF(QPointF(new_x, new_y), QPointF(x2, y2)))
+
+            elif self.resizing == "top_right":
+                # Check to see if outside the range of the original bounding box
+                if new_x > w:
+                    new_x = w
+                if new_x <= x1 + scaling_buffer + self.box_width:
+                    new_x = x1 + scaling_buffer + self.box_width
+                if new_y < 0:
+                    new_y = 0
+                if new_y >= y2 - scaling_buffer - self.box_width:
+                    new_y = y2 - scaling_buffer - self.box_width
+
+                # Update the bounding box
+                self.setRect(QRectF(QPointF(x1, new_y), QPointF(new_x, y2)))
+
+            elif self.resizing == "bottom_left":
+                # Check to see if outside the range of the original bounding box
+                if new_x < 0:
+                    new_x = 0
+                if new_x >= x2 - scaling_buffer - self.box_width:
+                    new_x = x2 - scaling_buffer - self.box_width
+                if new_y > h:
+                    new_y = h
+                if new_y <= y1 + scaling_buffer + self.box_width:
+                    new_y = y1 + scaling_buffer + self.box_width
+
+                # Update the bounding box
+                self.setRect(QRectF(QPointF(new_x, y1), QPointF(x2, new_y)))
+
+            elif self.resizing == "bottom_right":
+                # Check to see if outside the range of the original bounding box
+                if new_x > w:
+                    new_x = w
+                if new_x <= x1 + scaling_buffer + self.box_width:
+                    new_x = x1 + scaling_buffer + self.box_width
+                if new_y > h:
+                    new_y = h
+                if new_y <= y1 + scaling_buffer + self.box_width:
+                    new_y = y1 + scaling_buffer + self.box_width
+
+                # Update the bounding box
+                self.setRect(QRectF(QPointF(x1, y1), QPointF(new_x, new_y)))
```

Comment:
For all the `self.resizing == ...` conditionals, define the inside block as a local function and just recall the local function

Reply:
I realize the inside blocks are not exactly the same, but there is definitely a pattern to be exploited here.

Reply:
I was writing a function to handle it all, but I would end up writing so many if statements for the different out of bounds ranges that I felt it was reasonable to make all the logic explicit here.